predicators/train_refinement_estimator.py

```diff
@@ -0,0 +1,285 @@
+"""Scripts for collecting refinement data and for training learned refinement
+cost estimators.
+
+Example usage to generate data and train:
+    python predicators/train_refinement_estimator.py --env narrow_passage \
+        --approach refinement_estimation --refinement_estimator tabular \
+        --seed 0
+
+To save a generated dataset under a name different from the default:
+    python predicators/train_refinement_estimator.py --env narrow_passage \
+        --approach refinement_estimation --refinement_estimator tabular \
+        --seed 0 --data_file_name my_data_file.data
+
+To skip data collection and load a dataset file instead:
+    python predicators/train_refinement_estimator.py --env narrow_passage \
+        --approach refinement_estimation --refinement_estimator tabular \
+        --seed 0 --load_data
+
+To specify a dataset file name different from the default:
+    python predicators/train_refinement_estimator.py --env narrow_passage \
+        --approach refinement_estimation --refinement_estimator tabular \
+        --seed 0 --load_data --data_file_name my_data_file.data
+
+To skip training the approach:
+    python predicators/train_refinement_estimator.py --env narrow_passage \
+        --approach refinement_estimation --refinement_estimator tabular \
+        --seed 0 --skip_training
+"""
+
+import logging
+import os
+import sys
+import time
+from collections import defaultdict
+from pathlib import Path
+from typing import List, Set, cast
+
+import dill as pkl
+
+from predicators import utils
+from predicators.approaches import create_approach
+from predicators.approaches.refinement_estimation_approach import \
+    RefinementEstimationApproach
+from predicators.envs import BaseEnv, create_new_env
+from predicators.ground_truth_nsrts import get_gt_nsrts
+from predicators.option_model import _OptionModelBase, create_option_model
+from predicators.planning import PlanningFailure, PlanningTimeout, \
+    _MaxSkeletonsFailure, _skeleton_generator, _SkeletonSearchTimeout, \
+    run_low_level_search
+from predicators.settings import CFG
+from predicators.structs import NSRT, Metrics, ParameterizedOption, \
+    Predicate, RefinementDatapoint, Task, Type
+
+assert os.environ.get("PYTHONHASHSEED") == "0", \
```

Comment:
we can remove this, unlikely that someone is going to use this script and not `main.py`